akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala

```diff
@@ -111,6 +106,26 @@ class FramingSpec extends StreamSpec {
 
     }
 
+    "work with multiple delimiters" in {
+      val delimiters = List("\n", "\r\n", "FOO_a", "FOO_b").map(ByteString(_))
+      val testSequence = mutable.Buffer.empty[ByteString]
+      val testSequenceWithDelimiters = mutable.Buffer.empty[ByteString]
+      (1 to 5 * delimiters.length).foreach { i =>
+        val delimiter = delimiters(i % delimiters.length)
+        val seq = completeTestSequences(delimiter)
+        testSequence.appendAll(seq)
+        testSequenceWithDelimiters.appendAll(seq.map(_ ++ delimiter))
+      }
+      val f = Source(testSequenceWithDelimiters.toList)
+        .via(rechunk)
+        .via(Framing.delimiters(delimiters, 256))
+        .runWith(Sink.seq)
+
+      withClue(s"delimiters: $delimiters") {
+        f.futureValue should ===(testSequence.toVector)
+      }
+    }
```

Comment:
I think it would be good to have explicit coverage for multi-delimiters wrt chunk boundaries where the delimiters cross two ByteStrings chunks from upstream.